lib/util/trip/drive_card.dart

```diff
@@ -1,74 +1,141 @@
 import 'package:flutter/material.dart';
 import 'package:flutter_gen/gen_l10n/app_localizations.dart';
-import 'package:timelines/timelines.dart';
 
 import '../../drives/models/drive.dart';
 import '../../drives/pages/drive_detail_page.dart';
-import '../custom_timeline_theme.dart';
-import '../locale_manager.dart';
+import '../../rides/models/ride.dart';
+import '../own_theme_fields.dart';
+import '../supabase.dart';
 import 'trip_card.dart';
+import 'trip_card_state.dart';
+import 'trip_overview.dart';
 
 class DriveCard extends TripCard<Drive> {
   const DriveCard(super.trip, {super.key});
 
+  @override
+  State<DriveCard> createState() => _DriveCardState();
+}
+
+class _DriveCardState extends TripCardState<DriveCard> {
+  Drive? _drive;
+  bool _fullyLoaded = false;
+
+  @override
+  void initState() {
+    super.initState();
+
+    setState(() {
+      _drive = widget.trip;
+      trip = _drive;
+    });
+    loadDrive();
+  }
+
+  @override
+  void didUpdateWidget(DriveCard oldWidget) {
+    if (!trip!.equals(widget.trip)) {
+      loadDrive();
+    }
+    super.didUpdateWidget(oldWidget);
+  }
+
+  Future<void> loadDrive() async {
+    Map<String, dynamic> data = await SupabaseManager.supabaseClient.from('drives').select('''
+      *,
+      rides(
+        *,
+        rider: rider_id(*)
+      )
+    ''').eq('id', widget.trip.id).single();
+    if (mounted) {
+      setState(() {
+        _drive = Drive.fromJson(data);
+        trip = _drive;
+        _fullyLoaded = true;
+      });
+    }
+  }
+
+  @override
+  Widget buildRightSide() {
+    return TripOverview(super.trip!).buildSeatIndicator(context, trip!);
+  }
+
+  Color pickBannerColor() {
+    if (!_fullyLoaded) {
+      return Theme.of(context).cardColor;
+    } else if (_drive!.endTime.isBefore(DateTime.now())) {
+      if (_drive!.cancelled) {
```

Comment:
Hm I would either
1. always show disabled
2. or show success only for `approved` rides